service/controller/v5/blobclient/blob_client.go

```diff
@@ -0,0 +1,181 @@
+package blobclient
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2018-07-01/storage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+const (
+	resourceNotFoundError = "NotFound"
+	maxRetriesRequests    = 3
+)
+
+type Config struct {
+	ContainerName         string
+	GroupName             string
+	StorageAccountName    string
+	StorageAccountsClient *storage.AccountsClient
+}
+
+type BlobClient struct {
+	containerName         string
+	groupName             string
+	storageAccountName    string
+	storageAccountsClient *storage.AccountsClient
+
+	// containerURL is configured separately from the default
+	// parameters.
+	containerURL azblob.ContainerURL
+}
+
+func (c *BlobClient) Boot(ctx context.Context) error {
+	var containerURL azblob.ContainerURL
+
+	key, err := c.getAccountPrimaryKey(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	sc, err := azblob.NewSharedKeyCredential(c.storageAccountName, key)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	p := azblob.NewPipeline(sc, azblob.PipelineOptions{})
+	u, _ := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net", c.storageAccountName))
+	serviceURL := azblob.NewServiceURL(*u, p)
+	containerURL = serviceURL.NewContainerURL(c.containerName)
+	_, err = containerURL.GetProperties(ctx, azblob.LeaseAccessConditions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	c.containerURL = containerURL
+
+	return nil
+}
+
+func New(config Config) (*BlobClient, error) {
+	if config.ContainerName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ContainerName must not be empty", config)
+	}
+	if config.GroupName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.GroupName must not be empty", config)
+	}
+	if config.StorageAccountName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.StorageAccountName must not be empty", config)
+	}
+	if config.StorageAccountsClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.StorageAccountsClient must not be empty", config)
+	}
+
+	blobClient := &BlobClient{
+		containerName:         config.ContainerName,
+		groupName:             config.GroupName,
+		storageAccountName:    config.StorageAccountName,
+		storageAccountsClient: config.StorageAccountsClient,
+	}
+
+	return blobClient, nil
+}
+
+func (c *BlobClient) BlobExists(ctx context.Context, blobName string) (bool, error) {
+	blobURL := c.containerURL.NewBlockBlobURL(blobName)
+
+	_, err := blobURL.GetProperties(ctx, azblob.BlobAccessConditions{})
+	if IsBlobNotFound(err) {
+		return false, nil
+	}
+	if err != nil {
```

Comment:
I see this in a couple of places. Let's stick to `fmt`. 

```suggestion
	} else if err != nil {
```